dev/tools/localization/gen_l10n.dart

```diff
@@ -0,0 +1,412 @@
+    .replaceAll('@methodParameters', genMethodParameters(bundle, key, 'int').join(', '))
+    .replaceAll('@intlMethodArgs', methodArgs.join(',\n      '));
+}
+
+String genSupportedLocaleProperty(Set<LocaleInfo> supportedLocales) {
+  const String prefix = 'static const List<Locale> supportedLocales = <Locale>[ \n    Locale(''';
+  const String suffix = '),\n  ];';
+
+  String resultingProperty = prefix;
+  for (LocaleInfo locale in supportedLocales) {
+    final String languageCode = locale.languageCode;
+    final String countryCode = locale.countryCode;
+
+    resultingProperty += '\'$languageCode\'';
+    if (countryCode != null)
+      resultingProperty += ', \'$countryCode\'';
+    resultingProperty += '),\n    Locale(';
+  }
+  resultingProperty = resultingProperty.substring(0, resultingProperty.length - '),\n    Locale('.length);
+  resultingProperty += suffix;
+
+  return resultingProperty;
+}
+
+bool _isValidClassName(String className) {
+  // Dart class name cannot contain non-alphanumeric symbols
+  if (className.contains(RegExp(r'[^a-zA-Z\d]')))
+    return false;
+  // Dart class name must start with upper case character
+  if (className[0].contains(RegExp(r'[a-z]')))
+    return false;
+  // Dart class name cannot start with a number
+  if (className[0].contains(RegExp(r'\d')))
+    return false;
+  return true;
+}
+
+bool _isDirectoryReadableAndWritable(String statString) {
+  if (statString[0] == '-' || statString[1] == '-')
+    return false;
+  return true;
+}
+
+String _importFilePath(String path, String fileName) {
+  final String replaceLib = path.replaceAll('lib/', '');
+  return '$replaceLib/$fileName.dart';
+}
+
+Future<void> main(List<String> args) async {
+  final argslib.ArgParser parser = argslib.ArgParser();
+  parser.addOption('arb-dir', defaultsTo: path.join('lib', 'l10n'));
+  parser.addOption('template-arb-file', defaultsTo: 'app_en.arb');
+  parser.addOption('output-localization-file', defaultsTo: 'app_localizations');
+  parser.addOption('output-class', defaultsTo: 'AppLocalizations');
+  final argslib.ArgResults results = parser.parse(args);
+
+  final String arbPathString = results['arb-dir'];
+  final String outputFileString = results['output-localization-file'];
+
+  final Directory l10nDirectory = Directory(arbPathString);
+  final File templateArbFile = File(path.join(l10nDirectory.path, results['template-arb-file']));
+  final File outputFile = File(path.join(l10nDirectory.path, '$outputFileString.dart'));
+  final String stringsClassName = results['output-class'];
+
+  if (!l10nDirectory.existsSync())
+    exitWithError(
+      "The 'arb-dir' directory, $l10nDirectory, does not exist.\n"
+      'Make sure that the correct path was provided.'
+    );
+  final String l10nDirectoryStatModeString = l10nDirectory.statSync().modeString();
+  if (!_isDirectoryReadableAndWritable(l10nDirectoryStatModeString))
+    exitWithError(
+      "The 'arb-dir' directory, $l10nDirectory, is not readable or writable.\n"
```

Comment:
is not readable or writable => doesn't allow reading and writing

Reply:
Done